src/frontend/js/components/AddressesManagement/index.tsx

```diff
@@ -0,0 +1,519 @@
+    }
+  };
+
+  /**
+   * Update the provided address to promote it as main
+   *
+   * @param {Joanie.Address} address
+   */
+  const promoteAddress = (address: Joanie.Address) => {
+    if (!address.is_main) {
+      setError(undefined);
+      addresses.methods.update(
+        {
+          ...address,
+          is_main: true,
+        },
+        {
+          onError: () => setError(intl.formatMessage(messages.actionPromotion)),
+        },
+      );
+    }
+  };
+
+  /**
+   * Create a new address according to form values
+   * then update `selectedAddress` state with this new one.
+   * If `save` checkbox input is checked, the address is persisted
+   * otherwise it is only stored through the `selectedAddress` state.
+   *
+   * @param {AddressFormValues} formValues address fields to update
+   */
+  const handleCreate = async ({ save, ...address }: AddressFormValues) => {
+    setError(undefined);
+    if (save) {
+      await addresses.methods.create(address, {
+        onSuccess: handleSelect,
+        onError: () => setError(intl.formatMessage(messages.actionCreation)),
+      });
+    } else {
+      handleSelect({
+        id: LOCAL_BILLING_ADDRESS_ID,
+        is_main: false,
+        ...address,
+      });
+    }
+  };
+
+  /**
+   * Update the `editedAddress` with new values provided as argument
+   * then clear `editedAddress` state if request succeeded.
+   *
+   * @param {AddressFormValues} formValues address fields to update
+   */
+  const handleUpdate = async ({ save, ...newAddress }: AddressFormValues) => {
+    setError(undefined);
+    addresses.methods.update(
+      {
+        ...editedAddress!,
+        ...newAddress,
+      },
+      {
+        onSuccess: () => setEditedAddress(undefined),
+        onError: () => setError(intl.formatMessage(messages.actionUpdate)),
+      },
+    );
+  };
+
+  /**
+   * Prevent form to be submitted and clear `editedAddress` state.
+   */
+  const resetEditedAddress = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setEditedAddress(undefined);
+    return false;
+  };
+
+  const onSubmit = useMemo(() => {
+    const callback = editedAddress ? handleUpdate : handleCreate;
+    return handleSubmit(callback);
+  }, [editedAddress, handleSubmit]);
+
+  useEffect(() => {
+    reset(editedAddress || defaultValues);
+    setError(undefined);
+  }, [editedAddress]);
+
+  return (
+    <div className="AddressesManagement">
+      <button className="button button--tertiary" onClick={handleClose}>
+        <svg className="button__icon" role="img">
+          <use href="#icon-chevron-down" />
+        </svg>
+        <FormattedMessage {...messages.closeButton} />
+      </button>
+      {error && (
+        <Banner
+          message={intl.formatMessage(messages.error, { action: error })}
+          type={BannerType.ERROR}
+          rounded
+        />
+      )}
+      {addresses.items.length > 0 ? (
+        <section className="address-registered">
```

Comment:
component is a bit long, wouldn't it be possible to have one with registered addresses and another one to edit address

Reply:
<img src="https://user-images.githubusercontent.com/9265241/159676800-d2a6fb12-3588-4f56-9dfa-bcd59c80fd1c.png" width="150" />

Reply:
@carofun I moved the edition form into an `AddressForm` component then the registered address into a `RegisteredAddress` component. I keep the logic and methods to update address within the main `AddressesManagement`.